testing/smoketest/get_username.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -eu
+
+# The gh command is faster and not as hacky as the fallback solution
+if which gh >/dev/null; then
+  user_name=$(gh api user -q ".login")
+else
+  user_name=$(ssh -T git@github.com 2>&1|cut -d'!' -f1|cut -d' ' -f2)
```

Comment:
I'd prefer GH names because the local name may contain spaces and any code points (unlikely, but not necessarily utf-8 or ASCII). With the GH name we are IMO safe, as it is much stricter.

So using local git user names possibly requires a conversion.
The pro would be that it doesn't require network access - but without that, the smoketest doesn't work anyway.